daft/dataframe/dataframe.py

```diff
@@ -127,7 +124,7 @@ def explain(self, show_optimized: bool = False, simple=False) -> None:
         print(builder.pretty_print(simple))
 
     def num_partitions(self) -> int:
-        return self.__builder.num_partitions()
+        return self.__builder.to_physical_plan_scheduler().num_partitions()
```

Comment:
Honestly probably ok even if this takes O(10s) I think, especially for the many files case